Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -651,6 +651,22 @@ struct CLuaFunctionParser<ErrorOnFailure, ReturnOnFailure, Func> : CLuaFunctionP
         }
     }
 
+    // Tuples can be used to return multiple results
+    template <typename... Ts>
+    inline int PushResult(lua_State* L, const std::tuple<Ts...>& tuple)
+    {
+        // Call Push on each element of the tuple
+        std::apply([L](const auto&... value) { (lua::Push(L, value), ...); }, tuple);
+        return sizeof...(Ts);
```

Comment:
This won't work if there's an `optional` with no value.
Instead you could do:
```cpp
return std::apply([L](const auto&... value) { return (Push(L, value) + ...); }, tuple);
```

Reply:
That makes me wonder: Is there even a valid case for empty optionals being pushed? If not we could just drop it entirely and make Push return `void`

Reply:
Ah, actually I'm sorry. An optional (even if empty) results in a `nil` pushed onto the stack.

Reply:
Right... the one I saw was `CLuaArgument`. 
```
inline int Push(lua_State* L, const CLuaArgument& arg)
    {
        if (arg.GetType() == LUA_TNONE)
            return 0;
```
Not sure what to do with that...

Reply:
Ah, yes.. Well, this might break something if it's inside a table, or part of a tuple.
But you really shouldn't be using a tuple in that case anyways..

Reply:
It'll likely break all kinds of stuff if you attempt to push a `LUA_TNONE` anyway. Maybe just disallowing that and throwing on any such attempt might be a better solution. Would also clean up `Push` a lot.

Reply:
Added that now: `lua::Push` always pushes exactly one value. Pushing `LUA_TNONE` via  `CLuaArgument` will throw.